### PR TITLE
optimize density_prior_box_op.h in detect model for cpu

### PR DESCRIPTION
**Pull request by lidanqing-intel**
This PR improve performance of density_prior_box_op ~3 times faster 
- test machine:  SKX 8180.
- test script:  ./test_analyzer_resnet50 --infer_model=models/detect --gtest_filter=Analyzer_resnet50.profile_mkldnn --paddle_num_threads=4 --repeat=1 --batch_size=1 --profile

test=develop

### CODE REVIEW

**Comment by bingyanghuang** [comment]
I tested Danqing's PR on Xeon 2650 v4, repeat=1000. Benchmark results as below:

density_prior_box | Calls | Total
-- | -- | --
Baseline | 3000 | 1216.8ms
Optimization | 3000 | 406.877ms

**Comment by luotao1** [comment]
I tested Danqing's PR on Xeon 2620 v3, repeat=100. Benchmark results as below:

density_prior_box | Calls | Total
-- | -- | --
Baseline | 100 | 149ms
Optimization | 100 | 49ms

And the accuracy is no diff. Thanks very much!

File: paddle/fluid/operators/detection/density_prior_box_op.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+/* Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
```
Comment (luotao1) [inline paddle/fluid/operators/detection/density_prior_box_op.h]: please don't update 2018->2019. 2018 is the create day of this PR.

File: paddle/fluid/operators/detection/density_prior_box_op.h [hunk 2]
```diff
@@ -131,8 +137,14 @@ class DensityPriorBoxOpKernel : public framework::OpKernel<T> {
...
 
     auto e_vars = framework::EigenMatrix<T, Eigen::RowMajor>::From(*vars);
-
-    e_vars = var_et.broadcast(Eigen::DSizes<int, 2>(box_num, 1));
+#ifdef PADDLE_WITH_MKLML
+#pragma omp parallel for collapse(2)
+#endif
+    for (int i = 0; i < box_num; ++i) {
+      for (int j = 0; j < variances.size(); ++j) {
+        e_vars(i, j) = variances[j];
```
Comment (luotao1) [inline paddle/fluid/operators/detection/density_prior_box_op.h]: For line140-145, how about using `blas.VCOPY` to speed up?

**Comment by luotao1** [comment]
> you mean the original code file is 2018?

It's my typo, I mean the original code file is 2018. After you change it. I will approve and merge this PR.

**Comment by kbinias** [comment]
LGTM

**Comment by luotao1** [comment]
please do not `git push -f` after the review.

**Pull request merged**